src/renderer/qss/js/qssSettingButtons.js

```diff
@@ -118,4 +118,120 @@
+            },
+            item: {}
+        },
+        styles: {
+            redButton: "fl-qss-redButton"
+        },
+        modelRelay: {
+            "caption": {
+                target: "caption",
+                singleTransform: {
+                    type: "fluid.transforms.free",
+                    func: "gpii.qss.volumeButtonPresenter.getCaption",
+                    args: ["{that}.model.value", "{that}.model.messages", "{that}"]
+                }
+            }
+        },
+        modelListeners: {
+            caption: {
+                this: "{that}.dom.caption",
+                method: "text",
+                args: ["{change}.value"]
+            }
+        },
+        invokers: {
+            renderImage: {
+                funcName: "gpii.qss.volumeButtonPresenter.renderImage",
+                args: ["{that}.dom.image", "{that}.model.item.schema.image", "{arguments}.0"]
+            },
+            hideTitle: {
+                funcName: "gpii.qss.volumeButtonPresenter.hideTitle",
+                args: ["{that}.dom.title", "{arguments}.0"]
+            },
+            changeButtonColor: {
+                funcName: "gpii.qss.volumeButtonPresenter.changeButtonColor",
+                args: ["{that}.container", "{that}.options.styles.redButton", "{arguments}.0"]
+            }
+        }
+    });
+
+
+
+    /**
+     * Show or hide the title of the button.
+     * @param {jQuery} titleElem - The jQuery object corresponding to the title of the button.
+     * @param {Number} value - The state of the button.
+     */
+    gpii.qss.volumeButtonPresenter.hideTitle = function (titleElem, value) {
+        if (!value) {
+            titleElem.hide();
+        } else {
+            titleElem.show();
+        }
+    };
+
+    /**
+     * If available in the setting's schema, shows the specified image for the button when the value of setting is 0.
+     * @param {jQuery} imageElem - The jQuery object corresponding to the image of the button.
+     * @param {String} image - The path to the image.
+     * @param {Number} value - The state of the button.
+     */
+    gpii.qss.volumeButtonPresenter.renderImage = function (imageElem, image, value) {
+        if (image && !value) {
+            var maskImageValue = fluid.stringTemplate("url(\"%image\")", {
+                image: image
+            });
+
+            // use a mask image to avoid having 2 different images (one for when the
+            // button is activated and one when it is not)
+            imageElem.css("mask-image", maskImageValue);
+            imageElem.show();
+        } else {
+            imageElem.hide();
+        }
+    };
+
+    /**
+     * Returns the caption of the toggle button that needs to be shown below the button's
+     * title in case the state of the button is "on".
+     * In the case of the Volume widget, the caption message is shown only when the value is 0.
+     * @param {Number} value - The state of the button.
```

Comment:
How does the Number represent the state of the button? Mention the scale somewhere. Could even be worth issuing a JSDoc @typedef to make this clear